apps/documentation/src/components/MdxProvider/MdxTitle/MdxTitle.tsx

```diff
@@ -1,47 +1,43 @@
 import type { ElementType, JSX, ReactNode } from 'react'
+import { useState } from 'react'
 import { Title, type TitleProps } from '@mantine/core'
 
 export default function MdxTitle(props: TitleProps): JSX.Element {
+  const [headingId] = useState(getIdFrom(props.children))
```

Comment:
https://github.com/tuono-labs/tuono/pull/467#discussion_r1934522429

> I did a useState in my updated code - was that necessary?

It isn't necessary, 
we are dealing with mostly static content so just declare it as plain variable without the useState

```suggestion
  const headingId = getIdFrom(props.children)
```